File: Data Structures & Algorithms/max-consecutive-ones/submission-0.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        running_max, max_1s = 0, 0
        for num in nums:
            if num == 0:
                max_1s = max(running_max, max_1s)
                running_max = 0
            else:
                running_max += 1
        max_1s = max(running_max, max_1s)
        return max_1s
        

File: Data Structures & Algorithms/jump-game-ii/submission-0.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int jumps = 0, currentEnd = 0, farthest = 0;
        for (int i = 0; i < (int)nums.size() - 1; ++i) {
            farthest = max(farthest, i + nums[i]);
            if (i == currentEnd) {        // exhausted current jump's territory
                jumps++;
                currentEnd = farthest;    // commit to the best reach available
            }
        }
        return jumps;
    }
};